### ISSUE TITLE
`core.askpass could not be read`

### ISSUE THREAD

**Issue by matthiaskrgr** [rustsec/rustsec#989]
`$cargo audit`
```
    Fetching advisory database from `https://github.com/RustSec/advisory-db.git`
error: couldn't fetch advisory database: git operation failed: failed to fetch repo: core.askpass could not be read
```

Tried to get some more info with `--verbose` but no luck.

I removed the advisory db inside ´~/.cargo/` to see if it helps but apparently it doesn't and it is not being recloned :sweat_smile: 

`cargo-audit 0.18.1`

**Comment by Shnatsel** [comment]
Thank you for the report!

v0.17.6 will probably work, but I'd like to get to the bottom of this.

So the error is coming from gix, specifically here: https://github.com/Byron/gitoxide/blob/16cac4724347401d7dde2c8567b2f26420bf7216/gix/src/config/snapshot/credential_helpers.rs#L25-L26

@Byron any tips on how to diagnose this?

**Comment by Shnatsel** [comment]
> see the value of `core.askpass`.

Is this a git configuration option? If so, is there a `git` command to dump it?

**Comment by Byron** [comment]
`git config core.askpass` should do the trick.

**Comment by matthiaskrgr** [comment]
hmm, `git config core.askpass` just prints an empty line for me :sweat_smile:

**Comment by Byron** [comment]
Great, thanks a lot! It seems that `git` ignores this as unset string, and `gix` sees it as empty-string. I will investigate as that shouldn't cause interpolation to trigger.

In the mean-time, removing that should fix the issue, right?

**Comment by matthiaskrgr** [comment]
yup, when I comment it out, `cargo audit` works again :)  :tada:

**Comment by Byron** [comment]
My research revealed that indeed, `git` explicitly [ignores empty paths](https://github.com/git/git/blob/6807fcfedab84bc8cd0fbf721bc13c4e68cda9ae/prompt.c#L55) in the case of `core.askpass`, and in the next release, `gitoxide` will do the same.

It's a bit unfortunate that the error stack is truncated here as it should have pointed out that fact [somewhat more clearly](https://github.com/Byron/gitoxide/blob/02c4659984fa6423bc76cc4980a143edaba8ace0/gix-config-value/src/path.rs#L37).

In conclusion, this is an interesting case of the tug of war between being lenient and convenient, hiding errors, or bringing them to to everyones attention so they can be fixed. `git` seems to decide that on a case by case basis, an I guess it's OK to adapt to on a case-by-case basis as well.

**Comment by Shnatsel** [comment]
This should be resolved in the latest version of `cargo audit`. Please reopen if the issue persists.
